### ISSUE TITLE
`function-paren-newline` "multiline" errors for single arguments on their own line

### ISSUE THREAD

**Issue by TiddoLangerak** [eslint/eslint#9286]
**Tell us about your environment**

* **ESLint Version:**
v4.6.1
* **Node Version:**
v8.3.0
* **npm Version:**
v5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
// relevant section:
"function-paren-newline" : ["error", "multiline"],
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
function foo(
  param1
)

// or

foo(
  "some very long argument that I'd love to have on it's own line"
);

```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
    1:13  error  Unexpected newline after '('                                 function-paren-newline
    3:1   error  Unexpected newline before ')'                                function-paren-newline
    7:4   error  Unexpected newline after '('                                 function-paren-newline
    9:1   error  Unexpected newline before ')'                                function-paren-newline
```

Some elaboration: with the `multiline` setting it is only allowed to have newlines in function parenthesis if there are at least 2 arguments. I would expect it to also accept newlines in function parenthesis for single arguments as well (as long as it's done consistent). 

I'm not sure if this is actually a bug or intended behaviour, if it's the latter then an option to allow newlines for single-argument functions would be very welcome.

**Comment by not-an-aardvark** [comment]
Thanks for the report. This is working as intended, because there are no linebreaks between arguments in your function (since you have only one argument).

Could you clarify what behavior you would expect for a new option? It seems like allowing newlines for single-argument functions would defeat the purpose of the `multiline` option, but maybe there's something I'm missing.

**Comment by TiddoLangerak** [comment]
I'm sorry, I just realize I've been looking at the rule all wrongly. I wanted to enforce that "either all parameters are on their own line, or none are", but I now realize that's way beyond the scope of this rule. I thought this rule concerned whitespace between parameters as well, but it's only for whitespace around the parentheses  (obviously, given the name). So for my use case I guess I just need to use `function-paren-newline: consistent` and find some other rule to deal with newlines between arguments.

**Comment by jakezatecky** [comment]
@not-an-aardvark 

I would echo @TiddoLangerak's desires for a rule that merely enforces "either all parameters are on their own line, or none are".  The `multiline` option does _nearly_ everything needed for that, with the exception of single arguments. There are times when an individual argument may be a bit verbose and sometimes it is cleaner for that parameter to live on its own line (typically when I write my React test cases).

The `consistent` option, while permitting a single parameter on a newline, is insufficient, because it allows cases that would be unacceptable:

``` js
// Bad
function foo(bar,
  baz) {}

// Also bad
foo(
  bar, baz
);
```

Should I go ahead and create a new issue requesting an additional option to this rule or can we re-open this one?

**Comment by yvele** [comment]
@jakezatecky @not-an-aardvark I'm also very interrested with that feature, I strongly need it to write compact and readable unit tests

```js
it("Should work", () => {
  expect(
    Foo.bar({ baz: "BAZ", thing: "THING" }, argument2, argument3)
  ).toBe(
    "something"
  );
});
```

**Comment by not-an-aardvark** [comment]
Feel free to create a new issue. That said, keep in mind that you can always implement your preference as a custom rule without it being implemented as part of ESLint itself.

**Comment by steve-taylor** [comment]
If this rule doesn't concern itself with single-parameter functions, then it should not produce errors of the kind demonstrated in this issue.

**Comment by not-an-aardvark** [comment]
@steve-taylor Your premise is incorrect -- `function-paren-newline` treats single-parameter functions the same as it treats other types of functions. It doesn't explicitly ignore them.

**Comment by Conturbo** [comment]
This solution worked for me:
_.eslintrc.json_
```json
"rules": {
    "function-paren-newline": ["error", "consistent"],
  },
```

[This doc helped me](https://eslint.org/docs/rules/function-paren-newline)


This was the code causing the bug:
_Component.jsx_
```jsx
        <div>
          {preload.shows
            .filter(
              show =>
                `${show.title} ${show.description}`
                  .toUpperCase()
                  .indexOf(this.state.searchTerm.toUpperCase()) >= 0,
            )
            .map(show => <ShowCard key={show.imdbID} show={show} />)}
        </div>
```
